stateresolutionv2.go

```diff
@@ -439,62 +441,83 @@ func (r *stateResolverV2) getFirstPowerLevelMainlineEvent(event PDU) (
 // also apply them on top of the partial state. If they fail auth checks then
 // the event is ignored and dropped. Returns two lists - the first contains the
 // accepted (authed) events and the second contains the rejected events.
-func (r *stateResolverV2) authAndApplyEvents(events []PDU) {
+func (r *stateResolverV2) authAndApplyEvents(events ...PDU) {
+	addFromAuthEventsIfNotRejected := func(event PDU, eventType, stateKey string) {
+		for _, authEventID := range event.AuthEventIDs() {
+			if _, ok := r.isRejectedCache[authEventID]; !ok {
+				r.isRejectedCache[authEventID] = r.isRejectedFn(authEventID)
```

Comment:
To avoid having to lookup the `authEventID` right after setting it
```suggestion
                                isRejected := r.isRejectedFn(authEventID)
				r.isRejectedCache[authEventID] = isRejected
				if isRejected {
				    continue
				}
```